test/tests/utils/schema.test.js

```diff
@@ -0,0 +1,128 @@
+import mongoose from 'mongoose';
+import mongooseConnect from '../../utils/mongoose-connect';
+import { getNestedFieldType } from '../../../src/utils/schema';
+
+describe('schema', () => {
```

Comment:
Here you have the complete list on how you can define nested fields in a mongoose schema.

This tests ensure that, for every ways of defining your nested fields, we are able to introspect the structure